docs_src/design/ucr/0003-Extending-Device-Data.md

```diff
@@ -0,0 +1,106 @@
+Device Resources.
+
+
+### Existing solutions
+In Eaton's legacy products, these configuration and status extensions could be added to the channels of the Device in the central data store, exist as channel metadata, or be stored in separate data objects. This was fine and 
+efficient for a self-contained, monolithic solution, but is awkward when dividing the monolith into decoupled
+microservices.
+
+In EdgeX today, Devices and Resources can be added that are not related to the south-bound Device or to each other, 
+except perhaps by well-chosen Labels or Tags.
+
+
+### Requirements
+1. Standard EdgeX means exists to relate configuration and status Resources for Application Services to the
+originating south-bound Device data, on a per-Device basis.
+2. The configuration and status Resources for Application Services can be viewed as "extending" the originating
+south-bound Device data.
+3. Application Services, including a REST-based UI, can find and use these configuration and status Resources 
+from other Application Services in addition to the orignal south-bound Device data.
+
+Not a requirement: means of using or combining Resources from multiple 
+
+### Discussion
+
+If this UCR is approved, before an ADR can be written, we must choose one of two paths forward toward a solution.
+
+One path (the ideal one) is to add the extended resources to the south-bound device and its device profile.
+1. Pros:
+    - One source of truth in core-data and core-metadata for all Resources related to each Device.
+    - No special properties have to be added to relate the extended to the original Device Resources.
+2. Cons:
+    - The south-bound Device Service could become confused by Resource entries in its Device Profile that it ought to ignore, so it would not be backwards compatible.
+    - Potential contention as multiple Application Services try to update the Device Profile, adding their Resources.
+    - Would have to add some notation so that core-command could determine which service to address to Get or Set
+    the Device Resources (original and extended), so not backwards compatible.
+
+The alternative path is to create an "Extended" Device under each Application Service, which holds its extended configuration and status Resources.
+1. Pros:
+    - Good decoupling of these extended Resources under each Application Service.
+    - Core-command can use conventional means to know which Service to reach for Get and Set of these Resources.
+    - No changes necessary to south-bound Device Services, who don't know about these extended Resources.
+2. Cons:
+    - Need to add some means to each extended Device to indicate which south-bound Device it extends.
+    - Need to add or modify some core API(s), or their query options, to allow lookup of the original south-bound Device and all of its extended Devices.
+    - Application Services need to do extra work to pull in the extended Resources in addition to the south-bound Device Resources.
```

Comment:
Purpose of the ADR is to make this choice and document why one path was chosen over the other path. Thus IMO, this can be removed and addressed in the ADR.

Reply:
> Purpose of the ADR is to make this choice and document why one path was chosen over the other path. Thus IMO, this can be removed and addressed in the ADR.

I agree. The information can be kept in the PR (description/comments) but shouldn't be in the UCR.

Reply:
OK, I'll remove them and put them in the PR itself as comments (good suggestion).